core/trino-parser/src/main/java/io/trino/sql/ExpressionFormatter.java

```diff
@@ -363,7 +364,7 @@ protected String visitExists(ExistsPredicate node, Void context)
         @Override
         protected String visitIdentifier(Identifier node, Void context)
         {
-            if (!node.isDelimited()) {
+            if (!node.isDelimited() && !reservedIdentifiers().contains(node.getValue().toUpperCase(Locale.ENGLISH))) {
```

Comment:
Fixed. I must have missed something and didn't get `reserved` to work on my first attempt.

Reply:
Yeah, they could be now extracted as a separate PR.

Reply:
> However the optimizations for `reservedIdentifiers()` & `sqlKeywords()` are no longer relevant to this PR after the change. Do we still want to them? Revive #13739 maybe?

Then they are no longer useful, since other usages of these methods do not care about their cost.

Reply:
Removed the optimization commit.